zopkio/adhoc_deployer.py

```diff
@@ -135,6 +135,20 @@ def install(self, unique_id, configs=None):
         if executable is None:
           logger.error("executable was not provided for unique_id: " + unique_id)
           raise DeploymentError("executable was not provided for unique_id: " + unique_id)
+
+        #if the executable is in remote location copy to local machine
+        if (":" in executable):
+          remote_location_server = executable.split(":")[0]
+          remote_file_path = executable.split(":")[1] 
+          remote_file_name = os.path.basename(remote_file_path)
+          local_temp_file_name = "/tmp/" + remote_file_name
```

Comment:
why not take the directory as a config (you can default to /tmp) in case the file excedes the tmp quota, also since this is a tmp file you should delete it at the end of the function (use a try finally block)